packages/dashmate/src/listr/tasks/reindexNodeTaskFactory.js

```diff
@@ -0,0 +1,161 @@
+const {Listr} = require('listr2');
+const {Observable} = require('rxjs');
+const CoreService = require('../../core/CoreService');
+
+/**
+ * @param {DockerCompose} dockerCompose
+ * @param {startCore} startCore
+ * @param {stopNodeTask} stopNodeTask
+ * @param {waitForCoreStart} waitForCoreStart
+ * @param {waitForCoreSync} waitForCoreSync
+ * @param {createRpcClient} createRpcClient
+ * @param {renderServiceTemplates} renderServiceTemplates
+ * @param {writeServiceConfigs} writeServiceConfigs
+ * @param {configFileRepository} configFileRepository
+ * @param {configFile} configFile
+ * @return {reindexNodeTask}
+ */
+function reindexNodeTaskFactory(
+  dockerCompose,
+  startCore,
+  stopNodeTask,
+  waitForCoreStart,
+  waitForCoreSync,
+  createRpcClient,
+  renderServiceTemplates,
+  writeServiceConfigs,
+  configFileRepository,
+  configFile,
+) {
+  /**
+   * @typedef {reindexNodeTask}
+   * @param {Config} config
+   */
+  function reindexNodeTask(config) {
+    return new Listr([
+      {
+        title: 'Check services are not running',
+        enabled: () => config.get('core.reindex.enable'),
+        task: async () => {
+          const isRunning = await dockerCompose.isServiceRunning(config.toEnvs());
+
+          if (isRunning) {
+            throw new Error('Services is running, stop your nodes first');
+          }
+        },
+      },
+      {
+        title: 'Set reindex mode',
+        enabled: () => config.get('core.reindex.enable'),
+        task: async () => {
+          config.set('core.reindex.enable', 1);
+
+          // Write configs
+          configFileRepository.write(configFile);
+          const configFiles = renderServiceTemplates(config);
+          writeServiceConfigs(config.getName(), configFiles);
+        },
+      },
+      {
+        title: 'Start core',
+        task: async (ctx) => {
+          const {docker} = dockerCompose
+
+          let containerId = config.get('core.reindex.containerId', false);
+          let containerInfo;
+
+          if (containerId) {
+            try {
+              containerInfo = await docker.getContainer(containerId).inspect();
+            } catch (e) {
+              if (e.reason === 'no such container') {
+                containerId = null
+              }
+              throw e
+            }
+          }
+
+          if (!containerId) {
+            ctx.coreService = await startCore(config);
+            const containerInfo = await ctx.coreService.dockerContainer.inspect();
+
+            ctx.reindexContainerId = containerInfo.Id;
+            config.set('core.reindex.containerId', containerInfo.Id);
+            configFileRepository.write(configFile);
+
+            return;
+          }
+
+          ctx.reindexContainerId = containerInfo.Id;
+          ctx.coreService = new CoreService(
```

Comment:
Don't you need to start it?

Reply:
It is already started at this point, if it's not, it will start it at line 110